react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+
+class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = { caroNum: 1, intId: null };
+    this.arrowClickHandler = this.arrowClickHandler.bind(this);
+    this.iconClickHandler = this.iconClickHandler.bind(this);
+    this.IconItem = this.IconItem.bind(this);
+    this.IconList = this.IconList.bind(this);
+    this.picTimer = this.picTimer.bind(this);
+  }
+
+  picTimer(prevIntId) {
+    if (prevIntId) {
+      clearInterval(prevIntId);
+    }
+    const intId = setInterval(() => {
+      if (this.state.caroNum !== (this.props.pics.length)) {
+        this.setState({ caroNum: (this.state.caroNum + 1) });
+      } else if (this.state.caroNum === (this.props.pics.length)) {
+        this.setState({ caroNum: 1 });
+      }
+    }, 3000);
+    this.setState({ intId: intId });
+    return intId;
+  }
+
+  arrowClickHandler(event) {
+    if (event.target.className === 'fas fa-arrow-right' && this.state.caroNum !== (this.props.pics.length)) {
+      this.setState({ caroNum: (this.state.caroNum + 1) });
+    } else if (event.target.className === 'fas fa-arrow-right' && this.state.caroNum === (this.props.pics.length)) {
+      this.setState({ caroNum: 1 });
+    } else if (event.target.className === 'fas fa-arrow-left' && this.state.caroNum !== 1) {
+      this.setState({ caroNum: (this.state.caroNum - 1) });
+    } else if (this.state.caroNum === 1) {
+      this.setState({ caroNum: (this.props.pics.length) });
+    }
+    this.picTimer(this.state.intId);
+  }
```

Comment:
This is okay but it would be better to just create three separate click handlers, then add them directly to their corresponding elements. Having three simple separate methods is easier to debug and maintain than one complex method. Also with separate methods you don't have to rely on class names to determine what action to take.